sdk/src/main/java/com/commutestream/sdk/HttpClient.java

```diff
@@ -60,20 +62,35 @@ private void init() {
     @Override
     public void getAd(final AdRequest adRequest, final AdResponseHandler adHandler) {
         final long startTime = System.nanoTime();
-        TimeZone tz = TimeZone.getDefault();
-        String tzName = tz.getDisplayName(false, TimeZone.LONG);
 
-        Location loc = adRequest.getLocation();
         Set<AgencyInterest> agency_interests = adRequest.getAgencyInterests();
 
          HttpUrl.Builder urlBuilder = mBaseURL.newBuilder("/v2/banner")
-                .addQueryParameter("aaid", adRequest.getAAID())
-                .addQueryParameter("ad_unit_uuid", adRequest.getAdUnitUuid())
-                .addQueryParameter("timezone", tzName);
+                 .addQueryParameter("session_id", adRequest.getSessionID())
+                 .addQueryParameter("aaid", adRequest.getAAID())
+                 .addQueryParameter("ad_unit_uuid", adRequest.getAdUnitUuid())
+                 .addQueryParameter("timezone", adRequest.getTimezone());
+
+        Location loc = adRequest.getLocation();
+
+        //prevent resending the same location twice
+        if(loc == null){
+            Log.d("CS_SDK", "No location to send");
+        }
+        else if(lastLocationSentConfirmed == loc){
+            loc = null;
+            Log.d("CS_SDK", "Duplicate location detected - not sending");
```

Comment:
This is likely to spam the publishers while developing with log messages that I think only make sense to us correct? Same thing with the other log below in the else